packages/hash/api/src/collab/save.ts

```diff
@@ -126,6 +126,57 @@ const calculateSaveActions = async (
   const draftIdToPlaceholderId = new Map<string, string>();
   const draftIdToBlockEntities = new Map<string, DraftEntity<BlockEntity>>();
 
+  const shouldSetTextProperties = <T extends {}>(
```

Comment:
I've renamed this to `ensureNecessaryLegacyTextLink` to make it clearer what's going on here, and added a comment. I've also made its return type more sensible. I can't move it to outside of the function because it mutates something in local scope. I could make it an argument but I think that makes it even less clear what is going on.